lib/charms/synapse/v0/matrix_auth.py

```diff
@@ -0,0 +1,516 @@
+# Copyright 2024 Canonical Ltd.
+# Licensed under the Apache2.0. See LICENSE file in charm source for details.
```

Comment:
This might be shipped without the LICENSE file., so I'd go for 
```
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

Reply:
That's an interesting point and you made me have a look at other libs :)

So observability have a reference to a LICENSE file, while data platform do not for instance. We make a reference to a LICENSE file on the nginx route library.

```shell
~/dev/canonical
❯ head is-charms/synapse-operator/lib/charms/observability_libs/v0/juju_topology.py
# Copyright 2022 Canonical Ltd.
# See LICENSE file for licensing details.
"""## Overview.

This document explains how to use the `JujuTopology` class to
create and consume topology information from Juju in a consistent manner.

~/dev/canonical
❯ head is-charms/synapse-operator/lib/charms/data_platform_libs/v0/data_interfaces.py
# Copyright 2023 Canonical Ltd.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,

~/dev/canonical
❯ head is-charms/synapse-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py
# Copyright 2021 Canonical Ltd.
# See LICENSE file for licensing details.

~/dev/canonical
❯ head is-charms/synapse-operator/lib/charms/nginx_ingress_integrator/v0/nginx_route.py
# Copyright 2024 Canonical Ltd.
# Licensed under the Apache2.0. See LICENSE file in charm source for details.
```

I see charmcraft doesn't [care](https://github.com/canonical/charmcraft/blob/13a3e049fa8dd2e7f4a759e94f28448d637334f6/charmcraft/templates/charmlibs/new_library.py.j2), which makes sense, so maybe @erinecon and @gregory-schiano can chip in? 

Personally, I'm in favor of removing this and for consistency sake do a cleanup of our libs (like mentioned, at least nginx has that). I checked and bind-operator has that same reference.

Thanks for spotting! :eyeglasses:

Reply:
leaving as is for now, we can talk about it

Reply:
Oh, I missed your reply. This file is distributed independently from the charm via charmcraft, so referring to a license file that is not part of the software being distributed (the library and nothing else) doesn't make sense. The nginx-route lib is not maintained by our team, btw. It's maintained by observability.